### PR TITLE
dev-python/google-auth-oauthlib: Google Authentication Library

### PR DESCRIPTION
**Pull request by dargor**
Signed-off-by: Gabriel Linder <linder.gabriel@gmail.com>
Package-Manager: Portage-2.3.62, Repoman-2.3.12

### CODE REVIEW

File: dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
```
Comment (juippis) [inline dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild]: Did/could you test with python3.7?

File: dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+inherit distutils-r1
+
+DESCRIPTION="Google Authentication Library"
+HOMEPAGE="https://pypi.org/project/google-auth-oauthlib/"
```
Comment (juippis) [inline dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild]: Please also add the github page in here.

File: dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="
+	dev-python/google-auth[${PYTHON_USEDEP}]
+	>=dev-python/requests-oauthlib-0.7.0[${PYTHON_USEDEP}]
+"
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (juippis) [inline dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild]: distutils-r1 will handle dependency for setuptools in BDEPEND which is correct.

File: dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+"
```
Comment (juippis) [inline dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild]: I see upstream has python tests enabled, could you implement those in the ebuild?

File: dev-python/google-auth-oauthlib/metadata.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="project">
+    <email>python@gentoo.org</email>
```
Comment (juippis) [inline dev-python/google-auth-oauthlib/metadata.xml]: Don't add new random ebuilds to be maintained by the python project, but you will need to be the maintainer if you wish it to be in gentoo tree. Via proxy-maintainer project of course.

File: dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild (hunk 1 above)
Comment (dargor) [reply dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild]: The lib and my script using it both work fine in a 3.7 venv.

In tree, `dev-python/google-auth` and `dev-python/requests-oauthlib` do not have `python3_7` flag. Do you want me to bump this flag on all deps and this new lib now, or in a separate PR ?

File: dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild (hunk 4 above)
Comment (dargor) [reply dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild]: Sure, I'll pick a python ebuild with the `test` flag to learn how to do this and I'll add it.

File: dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild (hunk 1 above)
Comment (juippis) [reply dev-python/google-auth-oauthlib/google-auth-oauthlib-0.2.0.ebuild]: I'd say better to do it in a separate PR, not to make this too complicated. Plus it gives you time to test it's working properly. When added later, you can just modify the PYTHON_COMPAT and no revbump needed, unless you do other changes in the ebuild. So it's quite simple to add python3.7 later.

**Comment by dargor** [comment]
Adding v0.3.0.

**Comment by juippis** [comment]
Could you squash your commits into one, and make it follow the default guidance for git message style? The last commits message is wrong, for example, but the two before that are fine.

**Comment by dargor** [comment]
Sure, I'll take care of this and add version 0.4.0 too.

**Comment by juippis** [comment]
Thanks :) FYI I added `dev-python/click[${PYTHON_USEDEP}]` as RDEPEND due to not being able to run the tool without it, 
```
$ google-oauthlib-tool 
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.6/google-oauthlib-tool", line 11, in <module>
    load_entry_point('google-auth-oauthlib==0.4.0', 'console_scripts', 'google-oauthlib-tool')()
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 2843, in load_entry_point
    return ep.load()
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 2433, in load
    self.require(*args, **kwargs)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 2456, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/usr/lib64/python3.6/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'click' distribution was not found and is required by the application
```
https://github.com/googleapis/google-auth-library-python-oauthlib/blob/master/setup.py#L21

**Pull request closed without merging**